source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,453 @@
+        }
+
+        std::vector<Duration> captureDurations;
+        std::vector<Duration> allDurations;
+
+        for(size_t i = 0; i < numFrames; i++)
+        {
+            const auto beforeCapture = HighResClock::now();
+            const auto frame2D = camera.capture2D(settings);
+            const auto afterCapture = HighResClock::now();
+
+            captureDurations.push_back(afterCapture - beforeCapture);
+        }
+        allDurations.push_back(computeMedianDuration(captureDurations));
+        allDurations.push_back(computeAverageDuration(captureDurations));
+
+        printCapture2DResults(allDurations);
+    }
+
+    void benchmarkSave(Zivid::Camera &camera, const size_t numFrames)
+    {
+        printSaveHeader(numFrames);
+
+        auto frame = camera.capture();
+        frame.getPointCloud();
+
+        std::vector<Duration> allDurations;
+        std::vector<std::string> fileNames{ "Zivid3D.zdf", "Zivid3D.ply", "Zivid3D.pcd", "Zivid3D.xyz" };
+        for(const auto &fileName : fileNames)
+        {
+            std::vector<Duration> durationsPerFormat;
+            for(size_t j = 0; j < numFrames; j++)
+            {
+                const auto beforeSave = HighResClock::now();
+                frame.save(fileName);
+                const auto afterSave = HighResClock::now();
+
+                durationsPerFormat.push_back(afterSave - beforeSave);
+            }
+
+            allDurations.push_back(computeMedianDuration(durationsPerFormat));
+            allDurations.push_back(computeAverageDuration(durationsPerFormat));
+        }
+        printSaveResults(allDurations);
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        auto camera = getFirstCamera(zivid);
+
+        const size_t numConnects = 10;
+        const size_t numFrames3D = 20;
+        const size_t numFrames2D = 50;
+        const size_t numFramesSave = 10;
+
+        const size_t exposureTime = getMinExposureTime(camera.modelName());
+        const std::vector<size_t> exposureTimeOneFrame{ exposureTime };
+        const std::vector<size_t> exposureTimeTwoFrames{ exposureTime, exposureTime };
+        const std::vector<size_t> exposureTimeThreeFrames{ exposureTime, exposureTime, exposureTime };
+
+        const std::vector<unsigned int> oneIris{ 21U };
+        const std::vector<unsigned int> twoIrises{ 17U, 27U };
+        const std::vector<unsigned int> threeIrises{ 14U, 21U, 35U };
+
+        benchmarkConnect(camera, numConnects);
+        camera.connect();
+        benchmarkCapture3DAndFilters(camera, oneIris, exposureTimeOneFrame, numFrames3D);
+        benchmarkCapture3D(camera, makeSettingsVector(twoIrises, exposureTimeTwoFrames, false, false), numFrames3D);
+        benchmarkCapture3DAndFilters(camera, threeIrises, exposureTimeThreeFrames, numFrames3D);
+        benchmarkCapture2D(camera, makeSettings2D(exposureTime), numFrames2D);
+        benchmarkSave(camera, numFramesSave);
```

Comment:
Nice and clean main :)

Reply:
I agree, took a while but now it does look nice.